universal-application-tool-0.0.1/app/auth/saml/SamlCiviFormProfileAdapter.java

```diff
@@ -113,6 +108,38 @@ public SamlCiviFormProfileAdapter(
     return Optional.of(mergeCiviFormProfile(existingProfile.get(), profile));
   }
 
+  @VisibleForTesting
+  Optional<Applicant> getExistingApplicant(SAML2Profile profile) {
+    // authority_id is used as the unique stable key for users. This is unique and stable per
+    // authentication provider.
+    String authorityId =
+        getAuthorityId(profile)
+            .orElseThrow(
+                () -> new InvalidSamlProfileException("Unable to get authority ID from profile."));
+
+    return applicantRepositoryProvider
+        .get()
+        .lookupApplicantByAuthorityId(authorityId)
+        .toCompletableFuture()
+        .join();
+  }
+
+  protected Optional<String> getAuthorityId(SAML2Profile profile) {
+    // In SAML the user is uniquely identified by the issuer and subject claims.
+    // https://docs.oasis-open.org/security/saml-subject-id-attr/v1.0/cs01/saml-subject-id-attr-v1.0-cs01.html#_Toc536097226
+    //
+    // Wee combine the two to create the unique authority id.
+    String issuer = profile.getIssuerEntityID();
+    // Subject identifies the specific user in the issuer.
+    // Pac4j treats the subject as special, and you can't simply ask for the "sub" claim.
```

Comment:
update the comment "sub" is NameId, or perhaps Subject for the containing elements, in SAML I believe

Reply:
update "sub" here too please